src/builder/yara_rule_builder.cpp

```diff
@@ -176,8 +176,8 @@ YaraRuleBuilder& YaraRuleBuilder::withComment(const std::string& comment, bool m
  */
 YaraRuleBuilder& YaraRuleBuilder::withStringMeta(const std::string& key, const std::string& value)
 {
-	if (key == std::string{} || value == std::string{})
-		throw RuleBuilderError("Error: String-Meta key and value must be non-empty.");
+	if (key == std::string{})
```

Comment:
Tip: Here, we basically ask if `key` is empty. The more straightforward way would be to use `if (key.empty())`, which does exactly what we need and sounds better when read in English (_"if key is empty"_). Also, we will no longer need to create an empty string just to use it for comparison and then destroy it.

If you like the tip, feel free to commit it separately as it is not related to the present PR. I just wanted to point that out.

Reply:
Thank you for the tip, I totally agree. Even though it is not related to this PR, I will include it because it is a small change of code.